testsuite/tests/uart.rs

```diff
@@ -153,14 +223,17 @@ mod tests {
         assert!(matches!(c, Err(Error::Framing)));
 
         // provoke an error (this does not seem to be absolutely deterministic
-        // and we've seen multiple error variants in the wild)
+        // and we've seen multiple error variants in the wild, including
+        // receiving the wrong but valid character)
         unwrap!(nb::block!(tx_fast.write(b'a')));
         let c = nb::block!(rx_slow.read());
         defmt::info!("{}", c);
-        assert!(matches!(c, Err(Error::Framing) | Err(Error::Noise)));
+        assert!(matches!(c, Ok(240) | Err(Error::Framing) | Err(Error::Noise)));
     }
 
-    // TODO: Check the parity. But currently, there is no way to configure the parity
+    // TODO: Check the parity. There is only a loopback test with different
+    // parity parity settings. But we should check receiving data with wrong
+    // parity too.
```

Comment:
Seems like a good idea :)

Reply:
I've added a test for receiving data with wrong parity with https://github.com/stm32-rs/stm32f3xx-hal/pull/239/commits/e5c699772c7bda4d6f618165707ab7c2a335a596. Borrowing out/re-purposing the serial device felt cumbersome, so I just went for testing mismatching parity.

It feels to me that there will be some more elegant way. I thought about putting just the pins into the test state but I'm still inconclusive.